src/components/Assignment.js

```diff
@@ -48,6 +50,37 @@ class Assignment extends Component {
     console.log("Assignment.onRadioClick " + event.target.value);
     this.setState({selected: event.target.value});
   }
+    
+    // Add course
+    addAssignment = (courseId, assignmentName, dueDate) => {
+        const token = Cookies.get('XSRF-TOKEN');
+        
+        fetch(`${SERVER_URL}/gradebook/add`,
+              {
+              method: 'POST',
+              headers: { 'Content-Type': 'application/json',
+              'X-XSRF-TOKEN': token  },
+              body: JSON.stringify(courseId, assignmentName, dueDate)
```

Comment:
This may be a problem, you may be intending to stringify a javascript object, not call the method with multiple args.

According to the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify), valid syntax:
```js
JSON.stringify(value)
JSON.stringify(value, replacer)
JSON.stringify(value, replacer, space)
```

Reply:
Also, if I `console.log()` the 3 params to this method (`(courseId, assignmentName, dueDate)`, I see:
```
assignmentName: undefined
courseId: {courseId: '01-01-1901', assignmentName: '01-01-1901', dueDate: '01-01-1901'}
dueDate: undefined
```

so it looks like you're passing unexpected data to this method which means you need also need to check the method call in the `handleAdd()` method in `AddAssignment.js`.